src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +22,32 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        final int prime = 42;
+        int result = 1;
+        result = prime * result + (width == null ? 0 : width.hashCode());
+        result = prime * result + (color == null ? 0 : color.hashCode());
+        result = prime * result + (length == null ? 0 : length.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || !obj.getClass().equals(Rectangle.class)) {
+            return false;
+        }
+        Rectangle objectCast = (Rectangle) obj;
+        if (length == null && width == null && color == null && objectCast.color == null
+                && objectCast.width == null && objectCast.length == null) {
+            return hashCode() == objectCast.hashCode();
+        }
```

Comment:
Do you need to complare hashcodes in equals?

Reply:
let's add these checks below

Reply:
what about this comment?

Reply:
oh, didn't see this before, too much tabs.